src/test/java/io/castle/client/CastleMergeHttpTest.java

```diff
@@ -0,0 +1,77 @@
+package io.castle.client;
+
+import io.castle.client.model.AuthenticateAction;
+import io.castle.client.model.AuthenticateFailoverStrategy;
+import okhttp3.mockwebserver.MockResponse;
+import okhttp3.mockwebserver.RecordedRequest;
+import org.junit.Assert;
+import org.junit.Test;
+import org.springframework.mock.web.MockHttpServletRequest;
+
+import javax.servlet.http.HttpServletRequest;
+
+public class CastleMergeHttpTest extends AbstractCastleHttpLayerTest {
+
+    public CastleMergeHttpTest() {
+        super(new AuthenticateFailoverStrategy(AuthenticateAction.CHALLENGE));
+    }
+
+    @Test
+    public void mergeContextIsSendOnHttp() throws InterruptedException {
+
+        //Given
+        server.enqueue(new MockResponse().setResponseCode(200));
+        String id = "12345";
+
+        // And a mock Request
+        HttpServletRequest request = new MockHttpServletRequest();
+        // And a extra context
+        CustomExtraContext customExtraContext = new CustomExtraContext();
+        customExtraContext.setAddString("String value");
+        customExtraContext.setCondition(true);
```

Comment:
could you make sure to add an attribute that is part of the default context, to validate the merging strategy?

Reply:
saw this is done later. Ignore.

Reply:
ok